draft-ietf-quic-transport.md

```diff
@@ -322,175 +322,356 @@ all field sizes are in bits.  When discussing individual bits of fields, the
+
+* Octet 0: 0x84 (Flags indicating Long header and appropriate server Cleartext
+  packet type)
+* Octets 1-8: Connection ID (server-selected value)
+* Octets 9-12: Packet Number (low 4 octets, random 31-bit initial value)
+* Octets 13-16: Version (echoed)
+* Octets 17+: Payload
+
+Other Server Cleartext packets contain:
+
+* Octet 0: 0x85 (Flags indicating Long header and appropriate server Cleartext
+  packet type)
+* Octets 1-8: Connection ID (echoed)
+* Octets 9-12: Packet Number (low 4 octets, random 31-bit initial value)
+* Octets 13-16: Version (echoed)
+* Octets 17+: Payload
+
+The client MUST choose a random value and use it as the Connection ID until the
+server replies with a server-selected Connection ID. Server-selected Connection
+IDs are used after a successful handshake, cleanly distinguishing packets that
+use them from packets using client-selected Connection IDs. All packets
+including and following a Server Cleartext End-of-Handshake packet use a
+server-selected Connection ID, as described in {{connection-id}}.
+
+The payload of Cleartext packets contains frames, as described in {{frames}}.
+(TODO: Add hash before frames.)
+
+
+## Encrypted Packets {#encrypted-packet}
+
+Packets encrypted with either 0-RTT or 1-RTT keys may be sent with long headers.
+Different packet types explicitly indicate the encryption level for ease of
+decryption. These packets contain:
+* Octet 0: 0x86, 0x87 or 0x88 (Flags indicating long header and an encrypted
+  packet type)
+* Octets 1-8: Connection ID (client or server-selected, see {{connection-id}})
+* Octets 9-12: Packet Number (low 4 octets)
+* Octets 13-16: Version
+* Octets 17+: Encrypted Payload
+
+A flags octet of 0x86 indicates a 0-RTT packet. Key phases are used by the QUIC
+packet protection to identify the correct packet protection keys after the 1-RTT
+keys are established. The default initial value key phase is 0. See {{QUIC-TLS}}
+for more details.
+
+The encrypted payload is both authenticated and encrypted using packet
+protection keys. {{QUIC-TLS}} describes packet protection in detail.  After
+decryption, the plaintext consists of a sequence of frames, as described in
+{{frames}}.
 
 
-## Packet Numbers
+## Connection ID {#connection-id}
+
+QUIC connections are identified by their 64-bit Connection ID. All long headers
+contain a Connection ID. Short headers indicate the presence of a Connection ID
+using the CONNECTION_ID flag. When present, the Connection ID is in the same
+location in all packet headers, making it straightforward for middleboxes, such
+as load balancers, to locate and use it.
+
+When a connection is initiated, the client MUST choose a random value and use it
+as the Connection ID in all Cleartext packets and all 0-RTT Encrypted
+packets. The client's Connection ID is a suggestion to the server. The server
+echoes this value in all packets until the handshake is successful (see
+{{QUIC-TLS}}. In its final successful handshake completion packet, the server
```

Comment:
I had the same question as @MikeBishop here.  I believe that the answer is that the server sends the "Final" long packet type on all of the packets after ServerHello.  It would only echo the client's connection ID if it needed to send a Version Negotiation packet or a HelloRetryReqest.